src/algebra/field/defs.lean

```diff
@@ -143,6 +143,11 @@ instance smul_division_ring : has_smul ℚ K :=
 
 lemma smul_def (a : ℚ) (x : K) : a • x = ↑a * x := division_ring.qsmul_eq_mul' a x
 
+-- named this way as there is another one for the `smul` given by an algebra instance
```

Comment:
I think the other lemma can be deleted, this one is more general

Reply:
It seems not to be:

```lean
type mismatch, term
  rat.smul_one_eq_coe' m
has type
  @eq ?m_1 (@has_smul.smul rat ?m_1 (@rat.smul_division_ring ?m_1 ?m_2) m 1)
    (@coe rat ?m_1 (@coe_to_lift rat ?m_1 (@rat.cast_coe ?m_1 (@division_ring.to_has_rat_cast ?m_1 ?m_2))) m)
but is expected to have type
  @eq A
    (@has_smul.smul rat A
       (@algebra.to_has_smul rat A rat.comm_semiring (@ring.to_semiring A (@division_ring.to_ring A _inst_1)) _inst_2)
       m
       1)
    (@coe rat A (@coe_to_lift rat A (@rat.cast_coe A (@division_ring.to_has_rat_cast A _inst_1))) m)
```

Reply:
If I forcibly elaborate it with `@`, I get this:

```lean
  @eq A (@has_smul.smul rat A (@rat.smul_division_ring A _inst_1) m 1)
    (@coe rat A (@coe_to_lift rat A (@rat.cast_coe A (@division_ring.to_has_rat_cast A _inst_1))) m)
but is expected to have type
  @eq A
    (@has_smul.smul rat A
       (@algebra.to_has_smul rat A rat.comm_semiring (@ring.to_semiring A (@division_ring.to_ring A _inst_1)) _inst_2)
       m
       1)
    (@coe rat A (@coe_to_lift rat A (@rat.cast_coe A (@division_ring.to_has_rat_cast A _inst_1))) m)
```

Reply:
In fact, this is to be expected (I think) - the Q-algebra cast has no reason to be defeq by default, just it should only be that way if we make it so (which is why we added qsmul). The old lemma should guard against that being not the case, and also will syntactically match for `rw` and such

Reply:
In practice there are no Q-algebra instances with a smul action that is not defeq to this one; and syntactic matches don't matter inside typeclass arguments.

So I suspect the other lemma can just be dropped since `subsingleton_rat_module` is enough to get rid of weird module instances if they did appear.